sway-error/src/error.rs

```diff
@@ -2089,6 +2332,79 @@ impl ToDiagnostic for CompileError {
+                                if &error.span == erroneous_expression_span { // If the whole expression is erroneous.
+                                    "is"
+                                } else {
+                                    "contains"
+                                },
+                                if *erroneous_expression_kind == "parentheses" {
+                                    ""
+                                } else {
+                                    a_or_an(erroneous_expression_kind)
+                                },
+                                erroneous_expression_kind
+                            )
+                        ),
+                        hints: vec![
+                            if &error.span != erroneous_expression_span {
+                                Hint::info(
+                                    source_engine,
+                                    erroneous_expression_span.clone(),
+                                    format!("{} the contained {erroneous_expression_kind}.",
+                                        if *erroneous_expression_kind == "parentheses" {
+                                            "These are"
+                                        } else {
+                                            "This is"
+                                        }
+                                    )
+                                )
+                            } else {
+                                Hint::none()
+                            },
+                        ],
+                        help: vec![
+                            format!("{} cannot be {}an assignment target.",
+                                ascii_sentence_case(&erroneous_expression_kind.to_string()),
+                                if &error.span == erroneous_expression_span {
+                                    ""
+                                } else {
+                                    "a part of "
+                                }
+                            ),
+                            Diagnostic::help_empty_line(),
+                            "In Sway, assignment targets must be one of the following:".to_string(),
+                            format!("{}- Expressions starting with a mutable variable, optionally having", Indent::Single),
+                            format!("{}  array or tuple element accesses, struct field accesses,", Indent::Single),
+                            format!("{}  or arbitrary combinations of those.", Indent::Single),
+                            format!("{}  E.g., `mut_var` or `mut_struct.field` or `mut_array[x + y].field.1`.", Indent::Single),
+                            Diagnostic::help_empty_line(),
+                            format!("{}- Dereferencing of an arbitrary expression that results in a", Indent::Single),
+                            format!("{}  reference to mutable value.", Indent::Single),
```

Comment:
On this one, I was bouncing back and forth, adding and removing the indefinite article. It can even be seen in commits I think. At the end I've settled for no article, don't ask me why. Anyhow, after your remark, I've rolled the "a mutable value" out in all error messages.